testsuite/cli/src/client_proxy.rs

```diff
@@ -679,60 +677,85 @@ impl ClientProxy {
         }
     }
 
-    /// Waits for the next transaction for a specific address and prints it
-    pub fn wait_for_transaction(
-        &mut self,
-        account: AccountAddress,
-        sequence_number: u64,
-    ) -> Result<()> {
-        let mut max_iterations = 5000;
-        println!(
-            "waiting for {} with sequence number {}",
-            account, sequence_number
-        );
-        loop {
-            stdout().flush().unwrap();
-
-            match self
-                .client
-                .get_txn_by_acc_seq(account, sequence_number - 1, true)
-            {
-                Ok(Some(txn_view)) => {
-                    println!();
-                    if txn_view.vm_status == VMStatusView::Executed {
-                        println!("transaction executed!");
-                        if txn_view.events.is_empty() {
-                            println!("no events emitted");
-                        }
-                        break Ok(());
-                    } else {
-                        break Err(format_err!(
-                            "transaction failed to execute; status: {:?}!",
-                            txn_view.vm_status
-                        ));
-                    }
+    /// Wait for transaction, this function is not safe for waiting for a specific transaction,
+    /// should use wait_for_signed_transaction instead.
+    /// TODO: rename to wait_for_account_seq or remove
+    pub fn wait_for_transaction(&self, address: AccountAddress, seq: u64) -> Result<()> {
```

Comment:
Since now this isn't used, can we just delete it completely?

Reply:
it is used by smoke test suite, and in smoke test suite case it does not know the transaction details, only know account address and sequence.
may need think how to handle it better in smoke test, but probably worry about it later.

Reply:
Let me think about it, current change just keep it live to limit the scope of this PR.
Our smoke test suite uses it, I need think about whether we can update smoke test to avoid use this function and remove it.